src/hermes_claude_agent_sdk/content_events.py

```
@@ -0,0 +1,608 @@
+                and safe_canonical is not None
+                and previous_canonical != safe_canonical
+            ):
+                self._usage_malformed = True
+                continue
+            self._usage_models.add(safe_model)
+            if safe_canonical is not None:
+                self._usage_canonical_models[safe_model] = safe_canonical
+            else:
+                self._usage_canonical_models.setdefault(safe_model, None)
+
+    def _model_provenance(self) -> tuple[str | None, str | None, str]:
```

Ruff (0.16.3)

[warning] Too many return statements (9 > 6)

(PLR0911)